app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1030,6 +1030,19 @@ def validate(self, data):
         return data
 
     def update(self, experiment, validated_data):
+        if (
+            experiment.is_rollout
+            and not experiment.is_paused
+            and experiment.status == NimbusExperiment.Status.LIVE
+            and experiment.status_next is None
+            and experiment.publish_status == NimbusExperiment.PublishStatus.IDLE
+            and validated_data.get("publish_status")
+            != NimbusConstants.PublishStatus.REVIEW
```

Comment:
Ok so I changed this back from `== NimbusConstants.PublishStatus.IDLE` to `!= NimbusConstants.PublishStatus.REVIEW` because it can also = `IDLE` when it is an end enrollment/end experiment _rejection_. The good thing is that the test cases caught it immediately ;)

Reply:
The aggregate of all six fields together is the unique state we want:

<google-sheets-html-origin><style type="text/css"><!--td {border: 1px solid #cccccc;}br {mso-data-placement:same-cell;}--></style>

publish status | IDLE | publish status | DIRTY (IDLE)
-- | -- | -- | --
status | LIVE | status | LIVE
status next | None | status next | None

Reply:
wait wait wait. The validated_data won't necessarily have `status`, `publish_status`, or `status_next`. In this case, we are only updating `population_percent`: 

<img width="824" alt="image" src="https://user-images.githubusercontent.com/43795363/220478790-028d26e9-bdb2-424b-8fa4-b9675d50bd7c.png">

Reply:
So we can't rely on the uniqueness of those six fields. We have to structure it to include the fields that will be coming through `update`

Reply:
oh do we need this check  
```
validated_data.get("publish_status")
            != NimbusConstants.PublishStatus.REVIEW
``` 
since we are checking all status, status next and publish status?

Reply:
Yes, both "End Experiment" and "End Enrollment" go through this `update` (they both have `publish_status=REVIEW`)

Reply:
If we are making a live update, like the example above, all we will have in `validated_data` is changelog and population_percent

Reply:
<img width="413" alt="image" src="https://user-images.githubusercontent.com/43795363/220486783-1264833c-a932-454a-9e4b-2c0e760d7714.png">

If we were guaranteed to always have pub status, status next, and status in `validated_data`, we could use these six fields to determine if it should be dirty. Since we can't guarantee that, we have to rule out these other four cases based on what data comes through `update`:

<img width="859" alt="image" src="https://user-images.githubusercontent.com/43795363/220487054-374ab04f-6fcb-455c-9eba-5c6ce44a8fc0.png">